assets/js/components/Work/Tabs/FileSetModal.jsx

```diff
@@ -0,0 +1,309 @@
+      })
+      .then((url) => {
+        ingestFileSet({
+          variables: {
+            accession_number: data.accession_number,
+            workId,
+            role: data.role,
+            metadata: {
+              description: data.description,
+              label: data.label,
+              original_filename: currentFile.name,
+              location: s3Location(url),
+            },
+          },
+        });
+      })
+      .catch((error) => {
+        console.error("There was a problem with the S3 upload", error);
+        toastWrapper(
+          "is-danger",
+          `There was a problem with the S3 upload: ${error}`
+        );
+      });
+  };
+
+  // Handle file drop
+  const onDrop = React.useCallback((acceptedFiles) => {
+    setCurrentFile(acceptedFiles[0]);
+  }, []);
+  const {
+    getRootProps,
+    getInputProps,
+    isDragActive,
+    isDragReject,
+  } = useDropzone({
+    onDrop,
+    accept: "image/tiff, image/jpeg",
+    maxFiles: 1,
+  });
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setValues({ ...values, [name]: value });
+  };
+
+  return (
+    <div className={`modal ${isHidden ? "" : "is-active"}`}>
+      {urlLoading || (
+        <FormProvider {...methods}>
+          <form
+            onSubmit={methods.handleSubmit(onSubmit)}
+            data-testid="fileset-form"
+          >
+            <div className="modal-background"></div>
+
+            <div className="modal-card">
+              <header className="modal-card-head">
+                <p className="modal-card-title">Add FileSet to Work</p>
+                <button
+                  className="delete"
+                  aria-label="close"
+                  onClick={() => {
+                    handleFormReset();
+                  }}
+                ></button>
+              </header>
+              <section className="modal-card-body">
+                {isSubmitting ? (
+                  <UISkeleton rows={10} />
+                ) : (
+                  <div className="box">
+                    {error && (
+                      <div className="notification is-danger">
+                        <Error error={error} />
+                      </div>
+                    )}
+                    {urlError && (
+                      <div className="notification is-danger">
+                        Error retrieving presigned url
+                      </div>
+                    )}
+                    <UIFormField label="Accession number">
+                      <UIFormInput
+                        isReactHookForm
+                        required
+                        label="FileSet label"
+                        data-testid="fileset-accession-number-input"
+                        name="accession_number"
+                        placeholder="accession number"
+                        onChange={handleInputChange}
+                        value={values.accession_number}
```

Comment:
If we really need to set the `value` here, then I'd argue we probably don't need the React Hook Form baggage.  Any way to get this to work without controlling the input's value?    I could try to mess with it if you want.

Reply:
I struggled with that. I couldn't get it to retain the form values after a submit with error. (like duplicate accession number) unless I manually set the values. 

If you'd like to try - great.